diff --git a/lib/providers.js b/lib/providers.js
index 8ee0dbc9c2295851f57f3dfdd702b243de386df3..8d98ad58271a6633a722fbca27b872715bc48124 100644
--- a/lib/providers.js
+++ b/lib/providers.js
@@ -28,7 +28,7 @@ function normalizeOAuth(c) {
     if (c.issuer)
         c.wellKnown ?? (c.wellKnown = `${c.issuer}/.well-known/openid-configuration`);
     const authorization = normalizeEndpoint(c.authorization, c.issuer);
-    if (authorization && !authorization.url?.searchParams.has("scope")) {
+    if (authorization?.url?.searchParams.has("scope") === false) {
         authorization.url.searchParams.set("scope", "openid profile email");
     }
     const token = normalizeEndpoint(c.token, c.issuer);
@@ -51,11 +51,11 @@ function defaultProfile(profile) {
     };
 }
 function normalizeEndpoint(e, issuer) {
-    if (!e || issuer)
-        return;
+    if (!e) return e;
     if (typeof e === "string") {
         return { url: new URL(e) };
     }
+    if (!e.url) return e;
     // If v.url is undefined, it's because the provider config
     // assumes that we will use the issuer endpoint.
     // The existence of either v.url or provider.issuer is checked in